cmake/build_env_info.cmake

```diff
@@ -108,6 +108,10 @@ FUNCTION(check_git_setup)
     -P ${CURRENT_LIST_DIR}/build_env_info.cmake
     BYPRODUCTS ${post_configure_file})
 
+  if(NOT DEFINED CMAKE_CXX_STANDARD)
+    # FIXME Kokkos CXX version somehow being set later and C++11 is the minimum required for raw literal strings
+    set(CMAKE_CXX_STANDARD 17)
+  endif()
```

Comment:
Do we actually set `CMAKE_CXX_STANDARD` anywhere in Kokkos?
Here:
https://github.com/kokkos/kokkos/blob/22380c7c384b39b74bf357c0e7d070095078d067/cmake/kokkos_pick_cxx_std.cmake#L14-L18
only `KOKKOS_` variable gets set.

Reply:
Mostly thinking if this can have any side effects for projects using Kokkos 🤔

Reply:
This PR is very much not the place to insert such a change.

How are we enforcing suitably modern C++ support otherwise? Why is that insufficient here?

If we really need something special for the version info file, why not use `target_compile_features` to say what it needs?

Reply:
> This PR is very much not the place to insert such a change.

where do you want to do it?

> How are we enforcing suitably modern C++ support otherwise? Why is that insufficient here?

standard is not set and it yielded a compile error on my local build with AppleClang.

> If we really need something special for the version info file, why not use target_compile_features to say what it needs?

`target_compile_features` or `CMAKE_CXX_STANDARD` either works.  Want me to change?

Reply:
Please use target compile feature. With that, I have no objection

Reply:
Done